### DISCUSSION TITLE
Laravel not picking up .env

### DISCUSSION THREAD

**Discussion by tiagoapp** [Dokploy/dokploy#822] [General]
Any idea why laravel isn't picking up the environment variables?

I created a new App > Git:

Repository URL: https://github.com/laravel/laravel.git
Branch: 11.x
Path: /

Then input all the Laravel .env.example into the Environment variables
APP_NAME=Laravel
APP_ENV=local
APP_KEY=
APP_DEBUG=true
APP_TIMEZONE=UTC
APP_URL=http://localhost
etc
etc

But it never picks them up.. we can see on the logs

...
[laravel:warn] Your app configuration references the DYNAMODB_ENDPOINT environment variable, but it is not set.
[laravel:warn] Your app configuration references the DB_URL environment variable, but it is not set.
[laravel:warn] Your app configuration references the DB_URL environment variable, but it is not set.
[laravel:warn] Your app configuration references the MYSQL_ATTR_SSL_CA environment variable, but it is not set.
[laravel:warn] Your app configuration references the DB_URL environment variable, but it is not set.
[laravel:warn] Your app configuration references the MYSQL_ATTR_SSL_CA environment variable, but it is not set.
[laravel:warn] Your app configuration references the DB_URL environment variable, but it is not set.
[laravel:warn] Your app configuration references the DB_URL environment variable, but it is not set.
[laravel:warn] Your app configuration references the REDIS_URL environment variable, but it is not set.
[laravel:warn] Your app configuration references the REDIS_USERNAME environment variable, but it is not set.
[laravel:warn] Your app configuration references the REDIS_URL environment variable, but it is not set.
[laravel:warn] Your app configuration references the REDIS_USERNAME environment variable, but it is not set.
[laravel:warn] Your app configuration references the AWS_ACCESS_KEY_ID environment variable, but it is not set.
[laravel:warn] Your app configuration references the AWS_SECRET_ACCESS_KEY environment variable, but it is not set.
[laravel:warn] Your app configuration references the AWS_BUCKET environment variable, but it is not set.
[laravel:warn] Your app configuration references the AWS_URL environment variable, but it is not set.

**Comment by tiagoapp** [comment]
If i open the terminal and do php artisan key:generate i also get

 ErrorException 

  file_get_contents(/app/.env): Failed to open stream: No such file or directory

**Reply by Siumauricio** [reply]
What's the repository url to test it?

**Comment by MattEzekiel** [comment]
Any update on this? Same issue here

**Comment by CesarLuisGC** [comment]
**Solution**
I managed to solve it, although I still don't fully understand why Dokploy can't retrieve the environment variables. This was my solution, and it works fine. I should clarify that I'm using docker-compose.ym. What I noticed is that the Laravel application keeps looking for an .env file, so what I do is copy the variables added in Dokploy directly into a new file.

I have my project Dockerized. I create a .sh file which I call from my php.Dockerfile. My .sh file contains the following:

'#!/bin/sh
set -e

echo "=> Generating .env file from the environment..."
'# Delete the old .env file if it exists and regenerate it with all the variables
rm -f .env

'# Replace all the environment variables injected by Dockerploy
'# Use sed to enclose the values ​​in double quotes: KEY=VALUE -> KEY="VALUE"
'# This prevents errors with spaces like "devux .net" or special characters
printenv | grep -E '^(APP_|DB_|REDIS_|MAIL_|LOG_|FILESYSTEM_|QUEUE_|BROADCAST_|CACHE_|SESSION_|MEMCACHED_|AWS_|PUSHER_|MIX_|DEUX_|TRUSTED_)' | sed 's/=\(.*\)/="\1"/' > .env

'# Check the size for the log (without displaying sensitive content)
SIZE=$(wc -c < .env)
echo "=> Generated .env file ($SIZE bytes) with protected values."

echo "=> Verifying folder structure..."
mkdir -p storage/framework/cache/data
mkdir -p storage/framework/sessions
mkdir -p storage/framework/views
mkdir -p storage/logs
mkdir -p bootstrap/cache
mkdir -p public/uploads

echo "=> Environment ready. Starting process..."
exec "$@"

**Comment by maks-oleksyuk** [comment]
If this is a deployment using compose, try adding this to your service:

```yml
env_file:
  - .env
```

Dokploy will create a `.env` file with the settings specified in the UI and transfer them to the container after the next rebuild. If you change any value, you will need to redeploy.

Many containers have settings to ignore `.env` files (using `.dockerignore`) and do not transfer them to the container for security reasons.
